### PR TITLE
BGAppRefreshTask

### LINKED ISSUE #746: iOS: Change background refresh to use BGAppRefreshTask

### PR DESCRIPTION
**Pull request by stuartbreckenridge**
Fixes #746 

**BGAppRefreshTask**

This PR adds support for `BGAppRefreshTask` and removes existing code for `performFetch`. Feed refreshes are scheduled based on `AppDefaults.refreshInterval`. Refresh tasks are scheduled when the background task runs or when the app enters the background.

**NOTE**: When the task is triggered on the device from the debugger, it does not work. It runs fine when triggered by the device.

### CODE REVIEW

File: iOS/AppDelegate.swift [hunk 1]
```diff
@@ -105,75 +105,54 @@ class AppDelegate: UIResponder, UIApplicationDelegate, UISplitViewControllerDele
...
-			}
-			
-			DispatchQueue.main.async {
-				AccountManager.shared.syncArticleStatusAll() {
-					completeProcessing()
-				}
-				
-			}
-			
-		}
```
Comment (vincode-io) [inline iOS/AppDelegate.swift]: What this code does is that it *immediately* does a sync of the article status.  It doesn't do a full refresh.  We just want it to quickly and immediately sync the article status.

Do we need to change anything in applicationDidEnterBackground?  I don't see that it is using anything deprecated.

File: Frameworks/Account/Account.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -595,6 +595,7 @@
 			baseConfigurationReference = D511EEB6202422BB00712EC3 /* Account_target.xcconfig */;
 			buildSettings = {
 				CODE_SIGN_IDENTITY = "";
+				DEVELOPMENT_TEAM = VGR7X4HYA5;
```
[the remark this answers is not in the document]
Comment (heckj) [reply Frameworks/Account/Account.xcodeproj/project.pbxproj]: @vincode-io I'm going to add a note in README and/or Contributing to mention setting up the shared settings `.xcconfig` - it seems to bite anyone new to the project

[the remark this answers is not in the document]
Comment (vincode-io) [reply Frameworks/Account/Account.xcodeproj/project.pbxproj]: It's already in the main README.

File: iOS/AppDelegate.swift (hunk 1 above)
Comment (stuartbreckenridge) [reply iOS/AppDelegate.swift]: I've restored this code.

File: iOS/AppDelegate.swift [hunk 3]
```diff
@@ -263,9 +189,10 @@ class AppDelegate: UIResponder, UIApplicationDelegate, UISplitViewControllerDele
 	}
 	
 	@objc func userDefaultsDidChange(_ note: Notification) {
-		updateBackgroundRefreshInterval()
+		//updateBackgroundRefreshInterval()
```
[the remark this answers is not in the document]
Comment (stuartbreckenridge) [reply iOS/AppDelegate.swift]: Done.

File: iOS/AppDelegate.swift [hunk 4]
```diff
@@ -283,19 +210,93 @@ class AppDelegate: UIResponder, UIApplicationDelegate, UISplitViewControllerDele
...
+	
+	//	func updateBackgroundRefreshInterval() {
+	//		let refreshInterval = AppDefaults.refreshInterval
+	//		if refreshInterval == .manually {
+	//			UIApplication.shared.setMinimumBackgroundFetchInterval(UIApplication.backgroundFetchIntervalNever)
+	//		} else {
+	//			UIApplication.shared.setMinimumBackgroundFetchInterval(AppDefaults.refreshInterval.inSeconds())
+	//		}
+	//	}
+
```
[the remark this answers is not in the document]
Comment (stuartbreckenridge) [reply iOS/AppDelegate.swift]: Done.

File: iOS/AppDelegate.swift [hunk 5]
```diff
@@ -283,19 +210,93 @@ class AppDelegate: UIResponder, UIApplicationDelegate, UISplitViewControllerDele
...
 	
-	func sendReceivedArticlesUserNotification(newArticleCount: Int) {
+	/// Schedules a background app refresh based on `AppDefaults.refreshInterval`.
+	func scheduleBackgroundFeedRefresh() {
+		let request = BGAppRefreshTaskRequest(identifier: "com.ranchero.NetNewsWire.FeedRefresh")
+		request.earliestBeginDate = Date(timeIntervalSinceNow: AppDefaults.refreshInterval.inSeconds())
+		do {
+			try BGTaskScheduler.shared.submit(request)
+		} catch {
+			print("Could not schedule app refresh: \(error)")
```
[the remark this answers is not in the document]
Comment (stuartbreckenridge) [reply iOS/AppDelegate.swift]: It's now on os.log.

File: Frameworks/Account/Account.xcodeproj/project.pbxproj (hunk 2 above)
[the remark this answers is not in the document]
Comment (stuartbreckenridge) [reply Frameworks/Account/Account.xcodeproj/project.pbxproj]: This should not be an issue now.

**Pull request merged**